pkg/controller/controller_instance.go

```diff
@@ -1495,7 +1521,12 @@ func (c *controller) processProvisionFailure(instance *v1beta1.ServiceInstance,
 		return err
 	}
 
-	return err
+	// Need to vary return error depending on whether the worker should
+	// requeue this resource.
+	if failedCond == nil || shouldMitigateOrphan {
+		return errorMessage
```

Comment:
This is not strictly required as we have updated the status above, so the instance will come back to the queue anyway.
Keeping it as it is, as it was written that way before.

Reply:
Actually this is needed but the comment is misleading. We need to prevent resetting the rate limiter.
/cc @kibbles-n-bytes this is another place where we claim that we want to requeue the resource when what we really need is to prevent invoking the `Forget()` method.
I will update the comment.